spec/codecs/netflow_spec.rb

```diff
@@ -1 +1,208 @@
 require "logstash/devutils/rspec/spec_helper"
+require "logstash/codecs/netflow"
+
+describe LogStash::Codecs::Netflow do
+  subject do
+    LogStash::Codecs::Netflow.new.tap do |codec|
+      expect{codec.register}.not_to raise_error
+    end
+  end
+
+  let(:decode) do
+    [].tap do |events|
+      subject.decode(data){|event| events << event}
+    end
+  end
+
+  context "Netflow 5" do
+    let(:data) do
+      # this netflow raw data was produced with softflowd and captured with netcat
+      # softflowd -D -i eth0 -v 5 -t maxlife=1 -n 127.0.01:8765
+      # nc -k -4 -u -l 127.0.0.1 8765 > netflow5.dat
+      IO.read(File.join(File.dirname(__FILE__), "netflow5.dat"), :mode => "rb")
+    end
+
+    let(:json_events) do
+      events = []
+      events << <<-END
+        {
+          "@timestamp": "2015-05-02T18:38:08.280Z",
+          "netflow": {
+            "version": 5,
+            "flow_seq_num": 0,
+            "engine_type": 0,
+            "engine_id": 0,
+            "sampling_algorithm": 0,
+            "sampling_interval": 0,
+            "flow_records": 2,
+            "ipv4_src_addr": "10.0.2.2",
+            "ipv4_dst_addr": "10.0.2.15",
+            "ipv4_next_hop": "0.0.0.0",
+            "input_snmp": 0,
+            "output_snmp": 0,
+            "in_pkts": 5,
+            "in_bytes": 230,
+            "first_switched": "2015-06-21T11:40:52.280Z",
+            "last_switched": "2015-05-02T18:38:08.279Z",
+            "l4_src_port": 54435,
+            "l4_dst_port": 22,
+            "tcp_flags": 16,
+            "protocol": 6,
+            "src_tos": 0,
+            "src_as": 0,
+            "dst_as": 0,
+            "src_mask": 0,
+            "dst_mask": 0
+          },
+          "@version": "1"
+        }
+      END
+
+      events << <<-END
+        {
+          "@timestamp": "2015-05-02T18:38:08.280Z",
+          "netflow": {
+            "version": 5,
+            "flow_seq_num": 0,
+            "engine_type": 0,
+            "engine_id": 0,
+            "sampling_algorithm": 0,
+            "sampling_interval": 0,
+            "flow_records": 2,
+            "ipv4_src_addr": "10.0.2.15",
+            "ipv4_dst_addr": "10.0.2.2",
+            "ipv4_next_hop": "0.0.0.0",
+            "input_snmp": 0,
+            "output_snmp": 0,
+            "in_pkts": 4,
+            "in_bytes": 304,
+            "first_switched": "2015-06-21T11:40:52.280Z",
+            "last_switched": "2015-05-02T18:38:08.279Z",
+            "l4_src_port": 22,
+            "l4_dst_port": 54435,
+            "tcp_flags": 24,
+            "protocol": 6,
+            "src_tos": 0,
+            "src_as": 0,
+            "dst_as": 0,
+            "src_mask": 0,
+            "dst_mask": 0
+          },
+          "@version": "1"
+        }
+      END
+
+      events.map{|event| event.gsub(/\s+/, "")}
+    end
+
+    it "should decode raw data" do
```

Comment:
Yes, as a first test, I decided to threat the `decode` method as a black box and simply assert what gets out of it. I decided it was **good enough** as a **first test**, especially in the context of having previously **no test** AND having little/no knowledge of Netflow. It that test fails, *there is probably something wrong* with code you changed in the coded. Nothing more, nothing less. 

We all want perfection, but since this is realistically not possible, perfection is a general direction we aim to and progress in done in steps, it will not happen overnight. Also note that your definition of perfection is not necessarily the same as others. 

I still maintain that this test is fine as it is, and I have no intention on spending a disproportionate amount of energy trying to make the perfect test suite (to please the betterspecs advocates) given the fact that, as you say, it is all about trade off. I am **all about** writing better code, and I think this test  suite is way better than it was. So it's a good step in the right direction. 

*«Even if the result of your test are ok, then when you hit an error this test are harder to mantain»* well, there was no test before, so I think having *something* that fails is already really good. also for the maintenance, well, I think these assertions are so obvious, I don't see any long term maintenance problem and maintenance of the codec itself will be **a lot** easier with this imperfect test in place. 

If you have specific improvements that are relevant to the scope of this PR, which do not involve refactoring everything that is there, please feel free to comment. Otherwise I invite you to file an issue, or better a PR, with your suggestions on making this a better spec.